open_prices/common/openfoodfacts.py

```diff
@@ -107,3 +120,133 @@ def get_product_dict(product, flavor=Flavor.off) -> JSONType | None:
+            continue
+        product_code = product["code"]
+
+        # Skip duplicate products
+        if product_code in seen_codes:
+            continue
+        seen_codes.add(product_code)
+
+        # Some products have no "lang" field (especially non-OFF products)
+        product_lang = product.get("lang", product.get("lc", "en"))
+        # Store images & last_modified_t
+        product_images: JSONType = product.get("images", {})
+        product_last_modified_t = product.get("last_modified_t")
+
+        # Convert last_modified_t to a datetime object
+        # (sometimes the field is a string, convert to int first)
+        if isinstance(product_last_modified_t, str):
+            product_last_modified_t = int(product_last_modified_t)
+        product_source_last_modified = (
+            datetime.datetime.fromtimestamp(
+                product_last_modified_t, tz=datetime.timezone.utc
+            )
+            if product_last_modified_t
+            else None
+        )
+        # Skip products that have no last_modified date
+        if product_source_last_modified is None:
+            continue
+
+        # Skip products that have been modified today (more recent updates are
+        # possible)
+        if product_source_last_modified >= start_datetime:
+            print(f"Skipping {product_code}")
+            continue
+
+        # Build product dict to create/update
+        product_dict = {
+            key: product[key] if (key in product) else None for key in OFF_FIELDS
+        }
+        product_dict["image_url"] = generate_main_image_url(
+            product_code, product_images, product_lang, flavor=flavor
+        )
+        product_dict["source"] = flavor
+        product_dict["source_last_synced"] = timezone.now()
+        product_dict = normalize_product_fields(product_dict)
+
+        # Case 1: new OFF product (not in OP database)
+        if product_code not in existing_product_flavor_codes:
+            product_dict["code"] = product_code
+            products_to_create_or_update.append(Product(**product_dict))
+            added_count += 1
+
+        # Case 2: existing product (already in OP database)
+        else:
+            # Update the product if it:
+            # - is part of the current flavor sync (or if it has no source (created in Open Prices before OFF))  # noqa
+            # - has not been updated since the creation of the current dataset
+            existing_product_qs = (
+                Product.objects.filter(code=product_code)
+                .filter(Q(source=flavor) | Q(source=None))
+                .filter(
+                    Q(source_last_synced__lt=product_source_last_modified)
+                    | Q(source_last_synced=None)
+                )
+            )
+            if existing_product_qs.exists():
+                products_to_create_or_update.append(Product(**product_dict))
+                updated_count += 1
+
+        # update the database regularly
+        if (
```

Comment:
in the previous script we had `if buffer_len % batch_size == 0:`
but `0 % 1000 == 0` so it seems like it was commiting the db transactions for EVERY product and not in batch :sweat_smile: